homeassistant/components/openhome/update.py

```diff
@@ -0,0 +1,128 @@
+"""Update entities for Linn devices."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from homeassistant.components.update import (
+    UpdateDeviceClass,
+    UpdateEntity,
+    UpdateEntityFeature,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up update entities for Reolink component."""
+
+    _LOGGER.debug("Setting up config entry: %s", config_entry.unique_id)
+
+    device = hass.data[DOMAIN][config_entry.entry_id]
+
+    entity = OpenhomeUpdateEntity(device)
+
+    await entity.async_update()
+
+    async_add_entities([entity])
+
+
+class OpenhomeUpdateEntity(UpdateEntity):
+    """Update entity for a Linn DS device."""
+
+    def __init__(self, device):
+        """Initialize a Linn DS update entity."""
+        self._attr_device_class = UpdateDeviceClass.FIRMWARE
+        self._attr_supported_features = UpdateEntityFeature.INSTALL
+        self._name = None
+        self._device = device
+        self._installed_version = None
+        self._latest_version = None
+        self._release_summary = None
+        self._release_url = None
+
+    @property
+    def unique_id(self):
+        """Return a unique ID."""
+        return f"{self._device.uuid()}-update"
+
+    @property
+    def name(self):
+        """Return the name of the device."""
+        return f"{self._name} Firmware Update"
+
+    @property
+    def installed_version(self) -> str | None:
+        """Version currently in use."""
+        return self._installed_version
+
+    @property
+    def latest_version(self) -> str | None:
+        """Latest version available for install."""
+        return self._latest_version
+
+    @property
+    def release_summary(self) -> str | None:
+        """Description of the latest firmware release."""
+        return self._release_summary
```

Comment:
You are re-creating existing logic. See https://github.com/home-assistant/core/blob/dev/homeassistant/components/update/__init__.py#L210 etc

Reply:
Have refactored that - thanks. 

Is this the current best practice for entities? The openhome `media_player` entity follows a similar pattern of the `async_update` setting instance variables and a series of simple "getters" like `return self._volume_level`. If so, I'd like to refactor that out in a future PR?

Reply:
yeah, the best practice is to use the built-in `_attr_XXXX`